### PR TITLE
NAS-124047 / 24.04 / Refactor tables in SharesDashboardComponent

### PR DESCRIPTION
**Pull request by AlexKarpov98**
Testing: 
Check Shares route and ticket.

Result:
Screenshot 2023-09-13 at 14 08 57

### CODE REVIEW

File: src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-text/ix-cell-text.component.html [hunk 1]
```diff
@@ -1 +1 @@
-{{ value }}
+{{ value || ' - ' }}
```
Comment (bvasilenko) [inline src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-text/ix-cell-text.component.html:1]: Noticed that this change will affect all the falsy values. Examples of affected values are `false`, `""`, `0`, `null`, `undefined`. And difference between `""` and `undefined` (or between `0` and `null`) could play significant role in some configurations. I suggest to use a more strict check to display only for empty values (`null` or `undefined`)

File: src/app/modules/ix-table2/components/ix-table-body/ix-table-body.component.html [hunk 2]
```diff
@@ -6,7 +6,10 @@
     [hidden]="isLoading"
     (click)="onToggle(row)"
   >
-    <td *ngFor="let column of displayedColumns; index as idx">
+    <td
+      *ngFor="let column of displayedColumns; index as idx"
+      [ngClass]="column.cssClass ? column.cssClass : ''"
```
Comment (bvasilenko) [inline src/app/modules/ix-table2/components/ix-table-body/ix-table-body.component.html:11]: Maybe it would be easier to do `column.cssClass || ''`

**Review by bvasilenko — CHANGES_REQUESTED** [review]
In UI width of the heading needs a bit more work:

image

File: src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-text/ix-cell-text.component.html (hunk 1 above)
Comment (AlexKarpov98) [reply src/app/modules/ix-table2/components/ix-table-body/cells/ix-cell-text/ix-cell-text.component.html:1]: > Noticed that this change will affect all the falsy values. Examples of affected values are `false`, `""`, `0`, `null`, `undefined`. And difference between `""` and `undefined` (or between `0` and `null`) could play significant role in some configurations. I suggest to use a more strict check to display only for empty values (`null` or `undefined`)

okay, make sense

**Comment by AlexKarpov98** [comment]
> In UI width of the heading needs a bit more work:
> 
> image

@bvasilenko - can you help me reproducing this UI bug?
Which device size should I use?

Screenshot 2023-09-17 at 17 01 47

**Review by bvasilenko — APPROVED** [review]
LGTM

**Pull request merged**